### PR TITLE
Add needs_rotation to group responses

### PR DESCRIPTION
**Pull request by giarc3**
- Added needs_rotation as an `Option<bool>` to:
    - `GroupMetaResult`
    - `GroupGetResult`
    - `GroupBasicApiResponse`
    - `GroupGetApiResponse`
- `GroupListResult` is simply a `Vec<GroupMetaResult>`, so it will get these changes.
- Added a test for `group_get_metadata()` that tests a call by an admin, member, and non-member.

`needs_rotation` will be a `None` for a non-admin, and a `Some` for an admin.

### CODE REVIEW

File: tests/group_ops.rs [hunk 1]
```diff
@@ -18,16 +18,36 @@ fn group_create_no_member() {
...
+fn group_get_metadata() {
+    let admin_sdk = init_sdk();
+    let member_sdk = init_sdk();
+    let nonmember_sdk = init_sdk();
+
+    let member_id = member_sdk.device().account_id().clone();
+    let group = admin_sdk.group_create(&Default::default());
+    let group_id = &group.unwrap().id().clone();
+
+    let _ = admin_sdk.group_add_members(&group_id, &[member_id]);
```
Comment (coltfred) [inline tests/group_ops.rs]: I don't think you need the let _ =

File: src/internal/group_api/mod.rs [hunk 2]
```diff
@@ -172,6 +177,10 @@ impl GroupGetResult {
     pub fn member_list(&self) -> Option<&Vec<UserId>> {
         self.member_list.as_ref()
     }
+    /// None if the calling user is not a group admin
```
Comment (clintfred) [inline src/internal/group_api/mod.rs]: Should explain the other two states: Some(true), Some(false)

File: tests/group_ops.rs (hunk 1 above)
Comment (clintfred) [reply tests/group_ops.rs]: I think you will get a compiler warning about an unused Result if you ignore the result...

Comment (clintfred) [reply tests/group_ops.rs]: I see now that Craig used `?` instead. That's a good option. :+1:

File: src/internal/group_api/mod.rs (hunk 2 above)
Comment (giarc3) [reply src/internal/group_api/mod.rs]: I was/am having difficulty wording it succinctly. I made an attempt at a comment, but I am open to suggested changes.

File: src/internal/group_api/mod.rs [hunk 3]
```diff
@@ -123,6 +124,11 @@ impl GroupMetaResult {
     pub fn group_master_public_key(&self) -> &PublicKey {
         &self.group_master_public_key
     }
+    /// `Some(boolean)` indicating if the group needs rotation if the calling user is a group admin.
+    /// `None` if the calling user is not a group admin.
```
Comment (clintfred) [inline src/internal/group_api/mod.rs]: This is reasonable. My attempt that this documentation block would be:
```
/// `None` - calling user is not a group admin and cannot see if the group needs rotation or not
/// `Some(true)` - calling user is an admin and the group needs rotation
/// `Some(false)` - calling user is an admin and the group does not need rotation
```

**Pull request merged**